src/ott/geometry/geodesic.py

```diff
@@ -0,0 +1,282 @@
+      t = (jnp.sum(G) / jnp.sum(G > 0.)) ** 2
+
+    # Compute the coeffs of the Chebyshev pols approx using Bessel functs.
+    chebyshev_coeffs = compute_chebychev_coeff_all(
+        eigval, t, order, laplacian.dtype
+    )
+
+    return cls(
+        scaled_laplacian=scaled_laplacian,
+        eigval=eigval,
+        chebyshev_coeffs=chebyshev_coeffs,
+        t=t,
+        order=order,
+        **kwargs
+    )
+
+  def apply_kernel(
+      self,
+      scaling: jnp.ndarray,
+      eps: Optional[float] = None,
+      axis: int = 0,
+  ) -> jnp.ndarray:
+    r"""Apply :attr:`kernel_matrix` on positive scaling vector.
+
+    Args:
+      scaling: Scaling to apply the kernel to.
+      eps: passed for consistency, not used yet.
+      axis: passed for consistency, not used yet.
+
+    Returns:
+      Kernel applied to ``scaling``.
+    """
+    return expm_multiply(
+        self.scaled_laplacian, scaling, self.chebyshev_coeffs, self.eigval
+    )
+
+  @property
+  def kernel_matrix(self) -> jnp.ndarray:  # noqa: D102
+    n, _ = self.shape
+    kernel = self.apply_kernel(jnp.eye(n))
+    # check if the kernel is symmetric
+    if jnp.any(kernel != kernel.T):
+      kernel = (kernel + kernel.T) / 2.0
+    return kernel
+
+  @property
+  def cost_matrix(self) -> jnp.ndarray:  # noqa: D102
+    # Calculate the cost matrix using the formula (5) from the main reference
+    return -4 * self.t * mu.safe_log(self.kernel_matrix)
+
+  @property
+  def shape(self) -> Tuple[int, int]:  # noqa: D102
+    return self.scaled_laplacian.shape
+
+  @property
+  def is_symmetric(self) -> bool:  # noqa: D102
+    return True
+
+  @property
+  def dtype(self) -> jnp.dtype:  # noqa: D102
+    return self.scaled_laplacian.dtype
+
+  def transport_from_potentials(
+      self, f: jnp.ndarray, g: jnp.ndarray
+  ) -> jnp.ndarray:
+    """Not implemented."""
+    raise ValueError("Not implemented.")
+
+  def apply_transport_from_potentials(
+      self,
+      f: jnp.ndarray,
+      g: jnp.ndarray,
+      vec: jnp.ndarray,
+      axis: int = 0
+  ) -> jnp.ndarray:
+    """Since applying from potentials is not feasible in grids, use scalings."""
+    u, v = self.scaling_from_potential(f), self.scaling_from_potential(g)
+    return self.apply_transport_from_scalings(u, v, vec, axis=axis)
+
+  def marginal_from_potentials(
+      self,
+      f: jnp.ndarray,
+      g: jnp.ndarray,
+      axis: int = 0,
+  ) -> jnp.ndarray:
+    """Not implemented."""
+    raise ValueError("Not implemented.")
+
+  def tree_flatten(self) -> Tuple[Sequence[Any], Dict[str, Any]]:  # noqa: D102
+    return [
+        self.scaled_laplacian,
+        self.eigval,
+        self.chebyshev_coeffs,
+        self.t,
+        self.order,
+    ], {}
+
+  @classmethod
+  def tree_unflatten(  # noqa: D102
+      cls, aux_data: Dict[str, Any], children: Sequence[Any]
+  ) -> "Geodesic":
+    return cls(*children, **aux_data)
+
+
+def compute_largest_eigenvalue(laplacian_matrix, k, rng=None):
```

Comment:
Since this function only compute the largest eval, would remove `k` from the args and fix it to 1 in the code (so that user cannot accidentally compute more).